staging.Dockerfile

```diff
@@ -0,0 +1,74 @@
+# This Docker file tends to be used only
+# for `development` and `test` stage
+#
+# CLI command to run this file: script="some-name" docker-compose up --build
+# `some-name` is one of the names provided under `scripts` tag in `package.json` file
+# example: $ script="test" docker-compose up --build    --> will run telescope with `test` script
+# default: $ docker-compose up --build                  --> will run telescope with `start` script
+
+
+# Dockerfile
+#
+# ------------------
+# Context: Base Container
+FROM node:12.15.0-alpine3.9 AS base
+
+# Install Node
+RUN apk add --no-cache nodejs tini
+# Set Working Directory Context
+WORKDIR "/telescope"
+# Tini Entrypoint for Alpine
+ENTRYPOINT [ "/sbin/tini", "--"]
+
+COPY package.json .
+
+#
+# -----------------
+# Context: Build Context
+FROM node:12.15.0-alpine3.9 as build
+WORKDIR "/telescope"
+COPY package.json .
+
+# ------------------
+# Context: Dependencies
+FROM build AS dependencies
+
+# Install Production Modules
+RUN npm install --only=production --no-package-lock
+
+# Make Copy of Production Modules
+RUN cp -R node_modules prod_node_modules
+
+# Install Dev Modules (skip if we're not testing)
+RUN npm install --no-package-lock
```

Comment:
Why bother with this?  If we're going to do this, should it be gated on a flag we check from the env?